### PR TITLE
Core 445

### PR DESCRIPTION
**Pull request by hasahmad**

### CODE REVIEW

File: src/@commonsku/styles/ConfirmAddressPopup/ConfirmAddressContent.tsx [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+            const newAddress = selectedValue === 'recommended' ? {
+              ...(recommendedAddress || {}),
+              address_line_1: recommendedAddress?.address_line_1 || '',
+              address_line_2: '',
+              address_line_3: '',
+            } : address;
+            onSelect?.(newAddress || {});
+            onClose();
+          }}
+        >{!loading && !recommendedAddress && address.address_line_1 ? 'Make Changes' : 'Continue'}</Button>
```
Comment (robert-commonsku) [inline src/@commonsku/styles/ConfirmAddressPopup/ConfirmAddressContent.tsx:212]: It would be easier to read the code if the test for `!loading && !recommendedAddress && address.address_line_1` was stored in an appropriately named variable.

File: src/@commonsku/styles/ConfirmAddressPopup/ConfirmAddressContent.tsx
Comment (robert-commonsku) [inline src/@commonsku/styles/ConfirmAddressPopup/ConfirmAddressContent.tsx:1]: I think the confirm address stuff needs some refactoring.  Can you pull it out into a separate PR?

**Review by robert-commonsku — CHANGES_REQUESTED** [review]

File: src/@commonsku/styles/ConfirmAddressPopup/ConfirmAddressContent.tsx (hunk 1 above)
Comment (hasahmad) [reply src/@commonsku/styles/ConfirmAddressPopup/ConfirmAddressContent.tsx]: I've removed the confirm address popup components since they're not being used and likely will not be used

File: src/@commonsku/styles/ConfirmAddressPopup/ConfirmAddressContent.tsx
Comment (hasahmad) [reply src/@commonsku/styles/ConfirmAddressPopup/ConfirmAddressContent.tsx:1]: I've removed the confirm address popup components since they're not being used and likely will not be used

File: src/@commonsku/styles/AddressAutocomplete/AddressAutocompleteInput.tsx [hunk 2]
```diff
@@ -34,20 +34,15 @@ export default function AddressAutocompleteInput({
...
 
-  useEffect(() => {
-    if (typeof window === 'undefined') { return; }
-    const places = window.google.maps.places;
-    if (!places) { return; }
-    setSessionToken(
-      new places.AutocompleteSessionToken() || undefined
-    );
-  }, []);
+  const testId = useRef(uniqueId('autocomplete-search-input-'));
```
Comment (robert-commonsku) [inline src/@commonsku/styles/AddressAutocomplete/AddressAutocompleteInput.tsx:41]: I think that you can probably use the `useId()` hook for this. https://react.dev/reference/react/useId.

**Review by robert-commonsku — CHANGES_REQUESTED** [review]

File: src/@commonsku/styles/AddressAutocomplete/AddressAutocomplete.tsx [hunk 3]
```diff
@@ -68,17 +63,17 @@ export default function AddressAutocomplete({
         parsed_address: parseAddressComponents(res.address_components),
       });
     });
-  };
+  }, [onChange]);
 
-  const handleChange = debounce(handleChangeFn, delayValue);
+  const handleDebouncedChange = useDebounceCallback(handleChangeFn, delayValue);
 
   const loadOptions = debounce(async (value: string, callback: (options: readonly TOption[]) => void) => {
```
[the remark this answers is not in the document]
Comment (hasahmad) [reply src/@commonsku/styles/AddressAutocomplete/AddressAutocomplete.tsx:70]: I'm using this hook which should be memoized I think
https://usehooks-ts.com/react-hook/use-debounce-callback

**Review by robert-commonsku — APPROVED** [review]
